### PR TITLE
Fix primary email company tokens

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:
This PR fix wrong primary company tokens in email

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create company Test1 with address 1  Test1
3. Create company and assign company Test1
4. Create company Test2 with address 1 Test2
5. Then assign company Test2 to contact as primary but let also Test1 as secondary company
6. Create email with tokens
7. 
```
{contactfield=email}
{contactfield=companyname}
{contactfield=companyaddress1}
```

8. You should receive companyname and address1 from secondary company. Expect primary company data
9. After PR should work

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10733"></a>

### CODE REVIEW

**Review by escopecz — DISMISSED** [review]
The code makes sense. Nice test. Thanks Zdeno! 👍

File: app/bundles/LeadBundle/Helper/TokenHelper.php [hunk 1]
```diff
@@ -91,8 +91,12 @@ private static function getTokenValue(array $lead, $alias, $defaultValue)
         $value = '';
         if (isset($lead[$alias])) {
             $value = $lead[$alias];
-        } elseif (isset($lead['companies'][0][$alias])) {
-            $value = $lead['companies'][0][$alias];
+        } elseif (!empty($lead['companies'])) {
+            foreach ($lead['companies'] as $company) {
+                if (1 === (int) $company['is_primary'] && $company[$alias]) {
+                    $value = $company[$alias];
```
Comment (mmarton) [inline app/bundles/LeadBundle/Helper/TokenHelper.php:97]:
```suggestion
                    $value = $company[$alias];
                    break;
```

**Review by mmarton — CHANGES_REQUESTED** [review]
I've tried to test this, following the steps, but I didn't got any company info in the mails...
dumping the $lead variable from Tokenhelper  i have the following:

```
array:30 [
  "id" => "3"
  "title" => "test"
  "firstname" => null
  "lastname" => null
  "company" => "test 2"
  "position" => null
  "email" => "marton@m.hu"
  "mobile" => null
  "phone" => null
  "points" => 0.0
  "fax" => null
  "address1" => null
  "address2" => null
  "city" => null
  "state" => null
  "zipcode" => null
  "country" => null
  "preferred_locale" => null
  "timezone" => null
  "last_active" => "2022-01-23 20:07:39"
  "attribution_date" => null
  "attribution" => null
  "website" => null
  "facebook" => null
  "foursquare" => null
  "instagram" => null
  "linkedin" => null
  "skype" => null
  "twitter" => null
  "owner_id" => 1
]
```
Did I miss something?

**Comment by kuzmany** [comment]
@mmarton I suppose to  companies are fetch just for campaign/segment emails? Did you test these kind of emails?

File: app/bundles/LeadBundle/Helper/TokenHelper.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/LeadBundle/Helper/TokenHelper.php:97]: Make sense. Thanks

**Comment by mmarton** [comment]
No, I tried with the send email button on the top of the contact's page.

**Comment by kuzmany** [comment]
You have to test campaign or segment email

**Review by escopecz — APPROVED** [review]
The code looks good 👍

**Review by volha-pivavarchyk — APPROVED** [review]
Works as expected

**Pull request merged**